lib/ansible/modules/packaging/os/yum.py

```diff
@@ -934,6 +911,35 @@ def latest(module, items, repoq, yum_basecmd, conf_file, en_repos, dis_repos, in
                 pkgs['update'].append(spec)
                 will_update.add(spec)
                 continue
+
```

Comment:
I agree. Although, this fix solves an even more pressing issue: the state=latest option can't be used with url rpm as the module fails when the rpm is already installed.

After quickly reading the code (I wrote it a while ago), it seems to be independent. So if it can accelerate the reviewing process I can open another PR.